apps/web/components/ui/UsernameAvailability/PremiumTextfield.tsx

```diff
@@ -96,7 +96,7 @@ const PremiumTextfield = (props: ICustomUsernameProps) => {
   const updateUsername = trpc.viewer.updateProfile.useMutation({
     onSuccess: async () => {
       onSuccessMutation && (await onSuccessMutation());
-      await update();
+      await update({ username: inputUsernameValue });
```

Comment:
the update did not pass the new username value to update the session ( https://github.com/calcom/cal.com/pull/10019 )